### ISSUE TITLE
Leaking, part 2

### ISSUE THREAD

**Issue by rstacruz** [rstacruz/ractive-ractive#3]
here's a failing case thanks to @ElliotChong's work:

``` js
parent = new Ractive();
child = new Ractive();

child.set('data', 'datum');
parent.set('child', child);
expect(parent.get('child.data')).eql('datum'); // actually returns child.data (instance variable)
```

Need to uncover a few more isolated cases.

**Comment by rstacruz** [comment]
Here are some failing tests based on @ElliotChong's work again: aebc9358dba12b81f75f923d565d7ce623581785

**Comment by rstacruz** [comment]
AHA!

@elliotchong buddy, thanks so much for all your help. I nailed it down. The fix was _very_ simple. Let me know if `master` works for ya.

17564be6be3d8dee44a5925c5cf07b6dd6399d36

I've tested the tests you made in https://github.com/ElliotChong/ractive-ractive/commit/bfa9520e45beb40ca038a24517d22b52aa20b92c and they work just fine. I'm hesitant to commit it though: it's a gigantic test that should actually be broken up into smaller tests (it's really hard to follow what it actually does!).

edit: released as 1.1.1

**Comment by ElliotChong** [comment]
Thanks so much for your help @rstacruz!

Sorry for the delay in response, I've been a bit hectic with the holiday in the US.

So, I've finally isolated what the true issue is that I'm facing in my production code regarding leaking, give this test case a try and you should see the error:

```
    it('supports ractive children with the name data', function () {
      grandchild.set('data', 'Grandchild Data');
      child.set('data', grandchild);
      parent.set('child', child);

      expect(parent.get('child.data.data')).eql('Grandchild Data');
    });
```

It seems that assigning a Ractive instance as the name `data` to another Ractive instance exposes the leak, and I believe this will happen with other Ractive-reserved properties as well (clobbering `template` for example). For testing purposes however; it's really obvious with `data` since it will throw an Error.

**Comment by ElliotChong** [comment]
Just to be clear the leak appears to happen with all properties on the `child` / `grandchild` that have a Ractive instance as their value.
